flink-connector-mysql-cdc/src/main/java/com/ververica/cdc/connectors/mysql/source/split/MySqlSplitSerializer.java

```diff
@@ -114,15 +114,60 @@ public int getVersion() {
 
     @Override
     public MySqlSplit deserialize(int version, byte[] serialized) throws IOException {
-        if (version == VERSION) {
-            return deserializeV1(serialized);
+        switch (version) {
+            case 1:
+                return deserializeV1(serialized);
+            case 2:
+                return deserializeV2(serialized);
+            default:
+                throw new IOException("Unknown version: " + version);
         }
-        throw new IOException("Unknown version: " + version);
     }
 
     public MySqlSplit deserializeV1(byte[] serialized) throws IOException {
         final DataInputDeserializer in = new DataInputDeserializer(serialized);
 
+        int splitKind = in.readInt();
+        if (splitKind == SNAPSHOT_SPLIT_FLAG) {
+            TableId tableId = TableId.parse(in.readUTF());
+            String splitId = in.readUTF();
+            RowType splitKeyType = (RowType) LogicalTypeParser.parse(in.readUTF());
+            Object[] splitBoundaryStart = serializedStringToRow(in.readUTF());
+            Object[] splitBoundaryEnd = serializedStringToRow(in.readUTF());
+            BinlogOffset highWatermark = readBinlogPositionV1(in);
+            Map<TableId, TableChange> tableSchemas = readTableSchemas(in);
+
+            return new MySqlSnapshotSplit(
+                    tableId,
+                    splitId,
+                    splitKeyType,
+                    splitBoundaryStart,
+                    splitBoundaryEnd,
+                    highWatermark,
+                    tableSchemas);
+        } else if (splitKind == BINLOG_SPLIT_FLAG) {
+            String splitId = in.readUTF();
+            RowType splitKeyType = (RowType) LogicalTypeParser.parse(in.readUTF());
+            BinlogOffset startingOffset = readBinlogPositionV1(in);
+            BinlogOffset endingOffset = readBinlogPositionV1(in);
+            List<FinishedSnapshotSplitInfo> finishedSplitsInfo = readFinishedSplitsInfo(1, in);
+            Map<TableId, TableChange> tableChangeMap = readTableSchemas(in);
+            in.releaseArrays();
+            return new MySqlBinlogSplit(
+                    splitId,
+                    splitKeyType,
+                    startingOffset,
+                    endingOffset,
+                    finishedSplitsInfo,
+                    tableChangeMap);
+        } else {
+            throw new IOException("Unknown split kind: " + splitKind);
+        }
+    }
+
+    public MySqlSplit deserializeV2(byte[] serialized) throws IOException {
```

Comment:
Merge `deserializeV1` and `deserializeV2` and make `readBinlogPosition` accepts a version parameter?